src/nvim/ex_docmd.c

```diff
@@ -1369,29 +1264,43 @@ static char_u * do_one_cmd(char_u **cmdlinep,
    */
   save_cmdmod = cmdmod;
 
-  parse_state_from_global(&parsed);
-  parsed.eap = &ea;
-  parsed.verbose_save = -1;
-  parsed.save_msg_silent = -1;
-  parsed.did_esilent = 0;
-  parsed.did_sandbox = false;
-  bool parse_success = parse_one_cmd(cmdlinep, &parsed, cstack, fgetline, cookie);
```

Comment:
Is this removed because `parse_one_cmd` doesn't do as documented or is it useless or is it to make patching easier?

Reply:
Did you check https://github.com/neovim/neovim/pull/11340?

Reply:
I removed `parse_one_cmd` because it seems that the function does the same thing as the additional code from v8.1.0281 on `do_one_cmd`, except it brings no side-effect to the globals. The function is called only in one place though, so I wonder if making the function have no side-effect warrants the additional effort to port later patches. It also seems that the patch achieve the same goal as #11340.

That said, I'm not aware if the team have something in mind that requires the function to not do any side-effect. Let me know if I shouldn't remove it.

Reply:
Will have to revist this for recent Vim 8.2 patches such as https://github.com/vim/vim/commit/5661ed6c833e05467cab33cb9b1c535e7e5cc570.